### ISSUE TITLE
v1.0.2 breaks required properties array when extend is used 

### ISSUE THREAD

**Issue by kieranczer** [fastify/fluent-json-schema#73]
## 💥 Regression Report

Version 1.0.2 breaks the required properties array merge when extending a schema from another.

## Last working version

Worked up to version: 1.0.1

Stopped working in version: 1.0.2

## To Reproduce

```js
const S = require('fluent-schema');

const entityBase = S.object()
  .prop('id', S.string().format('uuid').required())
  .prop('version', S.string().format('date-time').required())
  .prop('label', S.string().required())
  .prop('createdBy', S.string().format('uuid').required())
  .prop('createdOn', S.string().format('date-time').required())
  .prop('modifiedBy', S.string().format('uuid').required())
  .prop('modifiedOn', S.string().format('date-time').required());

const test = S.object()
  .title('test schema')
  .additionalProperties(false)
  .prop('parent', S.string().format('uuid').required())
  .extend(entityBase);

module.exports = test;
```

## Expected behavior

In version 1.0.1 the required properties array in the json schema is produced correctly:

```json
"required": [
  "id",
  "version",
  "label",
  "createdBy",
  "createdOn",
  "modifiedBy",
  "modifiedOn",
  "parent"
]
```

In version 1.0.2 the required properties array gets messed up:

```json
"required": [
  {
    "0": "p",
    "1": "a",
    "2": "r",
    "3": "e",
    "4": "n",
    "5": "t"
  },
  "version",
  "label",
  "createdBy",
  "createdOn",
  "modifiedBy",
  "modifiedOn"
]
```

Likely a result of [this PR](https://github.com/fastify/fluent-schema/pull/69). 

Using version 1.0.1 until this is fixed 👍 
Btw, loving this package so far, keep up the awesome work! If I have some time I might look into fixing this.

## Your Environment

- *node version*: 12
- *os*: Mac

**Comment by aboutlo** [comment]
Thank you @kieranczer to report this, yes is related to #69. 
And if you have time to look in this would be cool :)

**Comment by aboutlo** [comment]
Resolved with #74
